code/modules/clothing/head/perceptomatrix.dm

```diff
@@ -0,0 +1,265 @@
+	clothing_traits = list(
+		/* eye/ear protection */
+		TRAIT_NOFLASH,
+		TRAIT_TRUE_NIGHT_VISION,
+		TRAIT_SIGHT_BYPASS,
+		TRAIT_EXPANDED_FOV,
+		TRAIT_GOOD_HEARING,
+		/* mental protection */
+		TRAIT_PERCEPTUAL_TRAUMA_BYPASS, //wip
+		TRAIT_RDS_SUPPRESSED,
+		TRAIT_MADNESS_IMMUNE,
+		/* psychic protection */
+		TRAIT_NO_MINDSWAP,
+		TRAIT_UNCONVERTABLE,
+	)
+	flags_cover = HEADCOVERSEYES|EARS_COVERED
+	flags_inv = HIDEHAIR|HIDEFACE
+	flash_protect = FLASH_PROTECTION_WELDER_SENSITIVE
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	equip_sound = 'sound/items/handling/helmet/helmet_equip1.ogg'
+	pickup_sound = 'sound/items/handling/helmet/helmet_pickup1.ogg'
+	drop_sound = 'sound/items/handling/helmet/helmet_drop1.ogg'
+	armor_type = /datum/armor/head_helmet_matrix
+	actions_types = list(/datum/action/cooldown/spell/pointed/percept_hallucination)
+
+	/// If we have a core or not
+	var/core_installed = FALSE
+	/// Active components to add onto the mob, deleted and created on core installation/removal
+	var/list/active_components
+
+// weaker overall but better against energy
+/datum/armor/head_helmet_matrix
+	melee = 15
+	bullet = 15
+	laser = 45
+	energy = 60
+	bomb = 15
+	fire = 50
+	acid = 50
+	wound = 10
+
+/obj/item/clothing/head/helmet/perceptomatrix/Initialize(mapload)
+	. = ..()
+
+	update_icon_state()
+	update_anomaly_state()
+
+	// some-fucking-how, RemoveElement fails to find the element in below proc if i try to remove it. so whatever.
+	AddElement(/datum/element/wearable_client_colour, /datum/client_colour/perceptomatrix, ITEM_SLOT_HEAD, forced = TRUE)
+
+/obj/item/clothing/head/helmet/perceptomatrix/equipped(mob/living/user, slot)
+	. = ..()
+	if(slot & ITEM_SLOT_HEAD)
+		RegisterSignal(user, COMSIG_MOB_BEFORE_SPELL_CAST, PROC_REF(pre_cast_core_check))
+
+/obj/item/clothing/head/helmet/perceptomatrix/dropped(mob/living/user, slot)
+	if(!(slot & ITEM_SLOT_HEAD))
+		UnregisterSignal(user, COMSIG_MOB_BEFORE_SPELL_CAST)
+	..()
+
+// Prevent casting the spell w/o the core.
+/obj/item/clothing/head/helmet/perceptomatrix/proc/pre_cast_core_check(mob/caster, datum/spell)
+	if(!core_installed && is_type_in_list(spell, actions_types))
+		to_chat(caster, "You can't zap minds without a core installd!")
+		return SPELL_CANCEL_CAST
+
+/obj/item/clothing/head/helmet/perceptomatrix/proc/update_anomaly_state()
+
+	// If the core isn't installed, or it's temporarily deactivated, disable special functions.
+	if(!core_installed)
+		clothing_flags = PERCEPTOMATRIX_INACTIVE_FLAGS
+		detach_clothing_traits(clothing_traits)
+		QDEL_LIST(active_components)
+		return
+
+	clothing_flags = PERCEPTOMATRIX_ACTIVE_FLAGS
+	attach_clothing_traits(initial(clothing_traits))
+
+	LAZYADD(active_components, AddComponent(/datum/component/wearertargeting/earprotection, list(ITEM_SLOT_HEAD)))
+	LAZYADD(active_components, AddComponent(
+		/datum/component/anti_magic, \
+		antimagic_flags = MAGIC_RESISTANCE_MIND, \
+		inventory_flags = ITEM_SLOT_HEAD, \
+	))
```

Comment:
Opening a second review just for clarity

> gotta instance the list somehow

So just instance it like any other list? lol

`var/list/active_components = list()`

Reply:
>:(